### PR TITLE
Gitlab Collector: Long running transactions collector and test

### PR DESCRIPTION
**Pull request by Sticksman**
Split from #819

### CODE REVIEW

File: collector/pg_long_running_transactions.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+var (
+	longRunningTransactionsCount = prometheus.NewDesc(
+		"pg_long_running_transactions",
+		"Current number of long running transactions",
+		[]string{},
+		prometheus.Labels{},
+	)
+
+	longRunningTransactionsAgeInSeconds = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, longRunningTransactionsSubsystem, "age_in_seconds"),
```
Comment (SuperQ) [inline collector/pg_long_running_transactions.go:46]:
```suggestion
		prometheus.BuildFQName(namespace, longRunningTransactionsSubsystem, "oldest_timestamp_seconds"),
```

File: collector/pg_long_running_transactions.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		"The current maximum transaction age in seconds",
+		[]string{},
+		prometheus.Labels{},
+	)
+
+	longRunningTransactionsQuery = `
+	SELECT
+		COUNT(*) as transactions,
+   		MAX(EXTRACT(EPOCH FROM (clock_timestamp() - xact_start))) AS age_in_seconds
```
Comment (SuperQ) [inline collector/pg_long_running_transactions.go:55]:
```suggestion
   		MAX(EXTRACT(EPOCH FROM clock_timestamp())) AS oldest_timestamp_seconds
```

File: collector/pg_long_running_transactions.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    FROM pg_catalog.pg_stat_activity
+    WHERE state is distinct from 'idle' AND (now() - xact_start) > '1 minutes'::interval AND query not like 'autovacuum:%'
```
Comment (SuperQ) [inline collector/pg_long_running_transactions.go:57]: I wonder if we shouldn't remove the `now() - xact_start` bit.

```suggestion
    WHERE state is distinct from 'idle' AND query not like 'autovacuum:%'
```

**Review by SuperQ — CHANGES_REQUESTED** [review]

File: collector/pg_long_running_transactions.go (hunk 2 above)
Comment (sberlin) [reply collector/pg_long_running_transactions.go]: How is this returning the oldest timestamp? Please take a look at #978

**Pull request merged**